From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Josip Sokcevic <sokcevic@chromium.org>
Date: Tue, 4 Jun 2024 23:07:34 +0000
Subject: Update clang-format location

The previous location clang-format GCS location was placing
content into CIPD package location (e.g. buildtools/linux64 for
buildtools/linux64/format). The new location removes that conflict (e.g.
places formatter into buildtools/linux64-format/ directory).

Bug: 336843583
Change-Id: Ia4228941fee2fde38df8f4c4f1cccfc41f366bf4
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5581721
Reviewed-by: Joanna Wang <jojwang@chromium.org>
Reviewed-by: Junji Watanabe <jwata@google.com>
Commit-Queue: Josip Sokcevic <sokcevic@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1310289}

diff --git a/DEPS b/DEPS
index 33eca22041b65c8d548b2b4c738d907dd0c652b6..e7c549e2b8a26d460f520eff1735d64e8b0ce52d 100644
--- a/DEPS
+++ b/DEPS
@@ -648,7 +648,7 @@ deps = {
       },
     ],
   },
-  'src/buildtools/win/format': {
+  'src/buildtools/win-format': {
     'bucket': 'chromium-clang-format',
     'condition': 'host_os == "win" and non_git_source',
     'dep_type': 'gcs',
@@ -662,7 +662,7 @@ deps = {
       },
     ],
   },
-  'src/buildtools/mac/format': {
+  'src/buildtools/mac-format': {
     'bucket': 'chromium-clang-format',
     'condition': 'host_os == "mac" and host_cpu == "x64" and non_git_source',
     'dep_type': 'gcs',
@@ -676,7 +676,7 @@ deps = {
       },
     ],
   },
-  'src/buildtools/mac_arm64/format': {
+  'src/buildtools/mac_arm64-format': {
     'bucket': 'chromium-clang-format',
     'condition': 'host_os == "mac" and host_cpu == "arm64" and non_git_source',
     'dep_type': 'gcs',
@@ -690,7 +690,7 @@ deps = {
       },
     ],
   },
-  'src/buildtools/linux64/format': {
+  'src/buildtools/linux64-format': {
     'bucket': 'chromium-clang-format',
     'condition': 'host_os == "linux" and non_git_source',
     'dep_type': 'gcs',
diff --git a/buildtools/.gitignore b/buildtools/.gitignore
index 449448ef5b822539e154d6f139058c567d0c2dec..caa2bd8ca4278415e707b39af7e731be24f4973c 100644
--- a/buildtools/.gitignore
+++ b/buildtools/.gitignore
@@ -3,16 +3,20 @@
 /android/doclava.tar.gz
 /android/doclava/
 /clang_format/script/
+/linux-format/
 /linux64/.versions/
 /linux64/clang-format
 /linux64/gn
+/mac-format/
 /mac/.versions/
 /mac/clang-format
 /mac/gn
+/mac_arm64-format/
 /reclient/
 /third_party/libc++/trunk
 /third_party/libc++abi/trunk
 /third_party/libunwind/trunk
+/win-format/
 /win/.versions
 /win/clang-format.exe
-/win/gn.exe
+/win/gn.exe
\ No newline at end of file
diff --git a/buildtools/DEPS b/buildtools/DEPS
index bc62395d0cdce163285bc5252397fea2f5d8b5fa..f32fc9148704879b0718936c5503e7834c7701ac 100644
--- a/buildtools/DEPS
+++ b/buildtools/DEPS
@@ -1,7 +1,13 @@
 use_relative_paths = True
 
+# Only these hosts are allowed for dependencies in this DEPS file.
+# If you need to add a new host, contact chrome infrastracture team.
+allowed_hosts = [
+  'chromium-clang-format',
+]
+
 deps = {
-  'win/format': {
+  'win-format': {
     'bucket': 'chromium-clang-format',
     'condition': 'host_os == "win"',
     'dep_type': 'gcs',
@@ -15,7 +21,7 @@ deps = {
       },
     ],
   },
-  'mac/format': {
+  'mac-format': {
     'bucket': 'chromium-clang-format',
     'condition': 'host_os == "mac" and host_cpu == "x64"',
     'dep_type': 'gcs',
@@ -29,7 +35,7 @@ deps = {
       },
     ],
   },
-  'mac_arm64/format': {
+  'mac_arm64-format': {
     'bucket': 'chromium-clang-format',
     'condition': 'host_os == "mac" and host_cpu == "arm64"',
     'dep_type': 'gcs',
@@ -43,7 +49,7 @@ deps = {
       },
     ],
   },
-  'linux64/format': {
+  'linux64-format': {
     'bucket': 'chromium-clang-format',
     'condition': 'host_os == "linux"',
     'dep_type': 'gcs',
diff --git a/docs/updating_clang_format_binaries.md b/docs/updating_clang_format_binaries.md
index e5cb28b5a037a68e48f53b9327f3f52f4b219a4c..03695427fe20a8e7042d83d31f506059e94296af 100644
--- a/docs/updating_clang_format_binaries.md
+++ b/docs/updating_clang_format_binaries.md
@@ -77,10 +77,10 @@ function upload_and_set {
 	set_deps $2 $object_info
 }
 
-upload_and_set buildtools/linux64/clang-format linux64/format
-upload_and_set buildtools/win/clang-format.exe win/format
-upload_and_set buildtools/mac/clang-format.x64 mac/format
-upload_and_set buildtools/mac/clang-format.arm64 mac_arm64/format
+upload_and_set buildtools/linux64/clang-format linux64-format
+upload_and_set buildtools/win/clang-format.exe win-format
+upload_and_set buildtools/mac/clang-format.x64 mac-format
+upload_and_set buildtools/mac/clang-format.arm64 mac_arm64-format
 
 # Clean up
 rm /tmp/clang-format-$CLANG_REV.tgz
@@ -100,7 +100,7 @@ clang-format differences by choosing patchset 1 as the base for the gerrit diff.
 ```shell
 ## New gerrit CL with results of old clang-format.
 # use old clang-format
-find base -name '*.cc' -o -name '*.c' -o -name '*.h' -o -name '*.mm' | xargs ./buildtools/linux64/format/clang-format -i
+find base -name '*.cc' -o -name '*.c' -o -name '*.h' -o -name '*.mm' | xargs ./buildtools/linux64-format/clang-format -i
 git commit -a
 git cl upload --bypass-hooks
 ## New patchset on gerrit CL with results of new clang-format.
